### ISSUE TITLE
Allow specification of network hardfork

### ISSUE THREAD

**Issue by wjmelements** [sc-forks/solidity-coverage#431]
`ganache-cli` supports a fork parameter, `-k`, which lets you specify which network upgrade to test with. It would be useful if there was a way to pass this information to the ganache. Perhaps a setting in .solcover.js?

**Comment by cgewecke** [comment]
@wjmelements 

I think if you want to test vs. `istanbul` you will need to upgrade to the [0.7.0 beta][1]. 

That lets you use any ganache version and specify [providerOptions][2] for it in .solcover.js. The options format follows ganache-core. In that context `-k` is called `hardfork`. 

(There's an upgrade guide from 0.6.x if you're interested, [here][4])

Please just lmk if you have any difficulty with this...

[1]: https://github.com/sc-forks/solidity-coverage/tree/beta#install
[2]: https://github.com/trufflesuite/ganache-core#options
[3]: https://github.com/sc-forks/solidity-coverage#options
[4]: https://github.com/sc-forks/solidity-coverage/blob/beta/docs/upgrade.md#upgrading-from-06x-to-070-betax

**Comment by wjmelements** [comment]
I'll do this Monday as part of TrustToken/registry#22

**Comment by wjmelements** [comment]
Thanks, I'll let you know how it goes.

**Comment by wjmelements** [comment]
I got it working. Here were the steps.

I installed the beta solidity-coverage and added the plugin line to truffle.js:

```
plugins: ["solidity-coverage"],
```


When I set
```
providerOptions: {
        "hardfork": "istanbul"
}
```
I get

```
Error: Hardfork istanbul not set as supported in supportedHardforks
```

So I install `ganache-cli` locally as:

```
"ganache-cli": "6.8.0-istanbul.0",
```

and set in `.solcover.js`

```
client: require('ganache-cli'),
```

I hope this will help others work through getting Istanbul working.

Thanks @cgewecke

**Comment by wjmelements** [comment]
Line coverage does not seem to work, though.

**Comment by cgewecke** [comment]
@wjmelements Sorry, could you elaborate on line coverage not working?

**Comment by cgewecke** [comment]
@wjmelements There's an issue in your project with how solc is configured in `truffle.js`. For internal reasons, solidity-coverage modifies the solc settings to compile without optimization, and assumes the Truffle V5 convention for that option. 

TLDR; removing the Truffle V4 style config: 
```javascript
/*solc: {
  version: "0.5.13",
  evmVersion: "istanbul",
  optimizer: {
    enabled: true,
    runs: 20000
  }
},*/
compilers: {
  solc: {
    version: "0.5.13",
    evmVersion: "istanbul",
    optimizer: {
      enabled: true,
      runs: 20000
    }
  },
},
```
... will give you this:
![Screen Shot 2019-11-18 at 7 04 46 PM](https://user-images.githubusercontent.com/7332026/69113134-f4375480-0a36-11ea-95ae-36c68e1c2aac.png)

**Comment by wjmelements** [comment]
Wow, thanks!
